pypesto/optimize/optimize.py

```diff
@@ -68,11 +68,12 @@ def minimize(
         optimizer = ScipyOptimizer()
 
     # startpoint method
-    if (startpoint_method is not None) \
-            and (problem.startpoint_method is not None):
-        raise Warning('Problem.startpoint_method will be ignored. Start '
-                      'points will be generated using the startpoint method '
-                      'given as an argument to the minimize function.')
+    if startpoint_method is not None and problem.startpoint_method is not None:
+        raise Warning(
+            "Problem.startpoint_method will be ignored. Startpoints will be "
+            "generated using the startpoint method given as an argument to "
+            "the minimize function.",
+        )
     elif problem.startpoint_method is not None:
```

Comment:
add a switch if `isinstance(startpoint_method, Callable)` that instantiates a `FunctionStartpoints` and issues a deprecation warning to mantain backwards compatibility?